packages/generators/app/lib/resources/templates/database-templates/js/database.template

```diff
@@ -0,0 +1,57 @@
+const path = require('path');
+
+module.exports = ({ env }) => {
+	const client = env('DATABASE_CLIENT', 'sqlite')
+
+	const connections = {
+		mysql: {
+			connection: {
+		    host: env('DATABASE_HOST', 'localhost'),
+		    port: env.int('DATABASE_PORT', 3306),
+		    database: env('DATABASE_NAME', 'strapi'),
+		    user: env('DATABASE_USERNAME', 'strapi'),
+		    password: env('DATABASE_PASSWORD', 'strapi'),
+		    ssl: env.bool('DATABASE_SSL', false) && {
+          key: env('DATABASE_SSL_KEY', undefined),
+          cert: env('DATABASE_SSL_CERT', undefined),
+          ca: env('DATABASE_SSL_CA', undefined),
+          capath: env('DATABASE_SSL_CAPATH', undefined),
+          cipher: env('DATABASE_SSL_CIPHER', undefined),
+          rejectUnauthorized: env.bool('DATABASE_SSL_REJECT_UNAUTHORIZED', true),
+        },
+			},
+		},
+		postgres: {
+			connection: {
+		    host: env('DATABASE_HOST', 'localhost'),
+		    port: env.int('DATABASE_PORT', 5431),
+		    database: env('DATABASE_NAME', 'strapi'),
+		    user: env('DATABASE_USERNAME', 'strapi'),
+		    password: env('DATABASE_PASSWORD', 'strapi'),
+		    ssl: env.bool('DATABASE_SSL', false) && {
+          key: env('DATABASE_SSL_KEY', undefined),
+          cert: env('DATABASE_SSL_CERT', undefined),
+          ca: env('DATABASE_SSL_CA', undefined),
+          capath: env('DATABASE_SSL_CAPATH', undefined),
+          cipher: env('DATABASE_SSL_CIPHER', undefined),
+          rejectUnauthorized: env.bool('DATABASE_SSL_REJECT_UNAUTHORIZED', true),
+        },
+			},
+	  },
+		sqlite: {
+			connection: {
+	      filename: path.join(__dirname, '..', env('DATABASE_FILENAME', 'data.db')),
+	    },
+	    useNullAsDefault: true,
+		}
+	}
+
+	const connection = env('DATABASE_URL') ? { connection: env('DATABASE_URL') } : connections[client]
```

Comment:
have you tested this?

Reply:
Not the DATABASE_URL part. It's part of the things I still need to do.
However the condition is tested in the e2e tests and the second part is correctly used.

Reply:
I forced the `DATABASE_URL` env variables during the test using a Postgres container. All the tests passed 👌

Now I just need to see how I could add the option to create a Strapi app using a connection string rather that db options. Because that could be used in the e2e tests to add one series of test. Which would ensure that we support connection strings as an option in the create-app CLI.

Reply:
I updated this part and did a lot of testing around it using the `getstarted`.

considering this config: 
```js
// Knex uses the strapi_test database
const postgres = {
  client: 'postgres',
  connection: {
    connectionString: 'postgresql://strapi_test:strapi_testlocalhost:5433/strapi_test',
    database: 'strapi',
    user: 'strapi',
    password: 'strapi',
    port: 5432,
    host: 'localhost',
  },
};
```

It default to the database set in the connection string.

Using an empty string as a connection string will ignore the connection string property:
```js
// Knex uses the strapi database
const postgres = {
  client: 'postgres',
  connection: {
    connectionString: '',
    database: 'strapi',
    user: 'strapi',
    password: 'strapi',
    port: 5432,
    host: 'localhost',
  },
};
```

This simplify the generated configuration and allow us to still be able to define configuration option if needed.